bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -1982,26 +1982,41 @@ private static boolean checkIsLikeTupleType(Object sourceValue, BTupleType targe
         }
 
         ArrayValue source = (ArrayValue) sourceValue;
-        if (source.size() != targetType.getTupleTypes().size()) {
+        if (targetType.getRestType() == null && source.size() != targetType.getTupleTypes().size()) {
             return false;
         }
 
+        List<BType> targetTypes = new ArrayList<>(targetType.getTupleTypes());
+        if (targetType.getRestType() != null) {
+            if (source.size() <= targetTypes.size()) {
+                return false;
+            } else {
+                int targetTypeSize = source.size() - targetTypes.size();
```

Comment:
We can remove the else block and move this out?

```suggestion
            }
            
            int targetTypeSize = source.size() - targetTypes.size();
```

Reply:
Changed